tests/test_aws.sh

```diff
@@ -729,14 +690,19 @@ EOF
   principal="$username"
   action="s3:ListBucketMultipartUploads"
   resource="arn:aws:s3:::$BUCKET_ONE_NAME"
-
   setup_user "$username" "$password" "user" || fail "error creating user"
 
   setup_bucket "s3api" "$BUCKET_ONE_NAME" || fail "error setting up bucket"
-  setup_policy_with_single_statement "$test_file_folder/$policy_file" "dummy" "$effect" "$principal" "$action" "$resource" || fail "failed to set up policy"
-  create_multipart_upload "$BUCKET_ONE_NAME" "$test_file" || fail "error creating multipart upload"
+  get_bucket_policy "s3api" "$BUCKET_ONE_NAME" || fail "error getting bucket policy"
+  log 5 "BUCKET POLICY: $bucket_policy"
+  get_bucket_acl "s3api" "$BUCKET_ONE_NAME" || fail "error getting bucket ACL"
+  log 5 "ACL: $acl"
+  setup_policy_with_single_statement "$test_file_folder/$policy_file" "dummy" "$effect" "$principal" "$action" "$resource"
+  assert_success "failed to set up policy"
+  create_multipart_upload "$BUCKET_ONE_NAME" "$test_file"
+  assert_success "failed to create multipart upload"
```

Comment:
For this assert to fire you need to wrap the 'create_multipart_upload' in a run - otherwise it will use the last run call's output/status and check that.